File: binding/include/printsupport/libqprinterinfo.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qprinterinfo.html
pub const qprinterinfo = struct {

    /// New constructs a new QPrinterInfo object.
    pub fn New() QtC.QPrinterInfo {
        return qtc.QPrinterInfo_new();
    }


    /// New2 constructs a new QPrinterInfo object.
    pub fn New2(other: ?*anyopaque) QtC.QPrinterInfo {
        return qtc.QPrinterInfo_new2(@ptrCast(other));
    }


    /// New3 constructs a new QPrinterInfo object.
    pub fn New3(printer: ?*anyopaque) QtC.QPrinterInfo {
        return qtc.QPrinterInfo_new3(@ptrCast(printer));
    }


    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QPrinterInfo_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    pub fn PrinterName(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPrinterInfo_PrinterName(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qprinterinfo.PrinterName: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Description(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPrinterInfo_Description(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qprinterinfo.Description: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Location(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPrinterInfo_Location(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qprinterinfo.Location: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn MakeAndModel(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPrinterInfo_MakeAndModel(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qprinterinfo.MakeAndModel: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn IsNull(self: ?*anyopaque, ) bool {
        return qtc.QPrinterInfo_IsNull(@ptrCast(self));
    }

    pub fn IsDefault(self: ?*anyopaque, ) bool {
        return qtc.QPrinterInfo_IsDefault(@ptrCast(self));
    }

    pub fn IsRemote(self: ?*anyopaque, ) bool {
        return qtc.QPrinterInfo_IsRemote(@ptrCast(self));
    }

    pub fn State(self: ?*anyopaque, ) i32 {
        return qtc.QPrinterInfo_State(@ptrCast(self));
    }

    pub fn SupportedPageSizes(self: ?*anyopaque, ) []const u8 {
        return qtc.QPrinterInfo_SupportedPageSizes(@ptrCast(self));
    }

    pub fn DefaultPageSize(self: ?*anyopaque, ) QtC.QPageSize {
        return qtc.QPrinterInfo_DefaultPageSize(@ptrCast(self));
    }

    pub fn SupportsCustomPageSizes(self: ?*anyopaque, ) bool {
        return qtc.QPrinterInfo_SupportsCustomPageSizes(@ptrCast(self));
    }

    pub fn MinimumPhysicalPageSize(self: ?*anyopaque, ) QtC.QPageSize {
        return qtc.QPrinterInfo_MinimumPhysicalPageSize(@ptrCast(self));
    }

    pub fn MaximumPhysicalPageSize(self: ?*anyopaque, ) QtC.QPageSize {
        return qtc.QPrinterInfo_MaximumPhysicalPageSize(@ptrCast(self));
    }

    pub fn SupportedResolutions(self: ?*anyopaque, ) []const u8 {
        return qtc.QPrinterInfo_SupportedResolutions(@ptrCast(self));
    }

    pub fn DefaultDuplexMode(self: ?*anyopaque, ) i32 {
        return qtc.QPrinterInfo_DefaultDuplexMode(@ptrCast(self));
    }

    pub fn SupportedDuplexModes(self: ?*anyopaque, ) []const u8 {
        return qtc.QPrinterInfo_SupportedDuplexModes(@ptrCast(self));
    }

    pub fn DefaultColorMode(self: ?*anyopaque, ) i32 {
        return qtc.QPrinterInfo_DefaultColorMode(@ptrCast(self));
    }

    pub fn SupportedColorModes(self: ?*anyopaque, ) []const u8 {
        return qtc.QPrinterInfo_SupportedColorModes(@ptrCast(self));
    }

    pub fn AvailablePrinterNames() []const u8 {
        return qtc.QPrinterInfo_AvailablePrinterNames();
    }

    pub fn AvailablePrinters() []const u8 {
        return qtc.QPrinterInfo_AvailablePrinters();
    }

    pub fn DefaultPrinterName(, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPrinterInfo_DefaultPrinterName();
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qprinterinfo.DefaultPrinterName: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn DefaultPrinter() QtC.QPrinterInfo {
        return qtc.QPrinterInfo_DefaultPrinter();
    }

    pub fn PrinterInfo(printerName: []const u8) QtC.QPrinterInfo {
        const printerName_str = qtc.libqt_string{
    .len = printerName.len,
    .data = printerName.ptr,
};
return qtc.QPrinterInfo_PrinterInfo(printerName_str);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QPrinterInfo_Delete(@ptrCast(self));
    }
};

File: binding/src/printsupport/libqprinterinfo.cpp
#include <QPageSize>
#include <QPrinter>
#include <QPrinterInfo>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qprinterinfo.h>
#include "libqprinterinfo.h"
#include "libqprinterinfo.hxx"

QPrinterInfo* QPrinterInfo_new() {
	 return new QPrinterInfo();
}

QPrinterInfo* QPrinterInfo_new2(const QPrinterInfo* other) {
	 return new QPrinterInfo(*other);
}

QPrinterInfo* QPrinterInfo_new3(const QPrinter* printer) {
	 return new QPrinterInfo(*printer);
}

void QPrinterInfo_OperatorAssign(QPrinterInfo* self, const QPrinterInfo* other) {
	self->operator=(*other);
}

libqt_string QPrinterInfo_PrinterName(const QPrinterInfo* self) {
	QString _ret = self->printerName();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QPrinterInfo_Description(const QPrinterInfo* self) {
	QString _ret = self->description();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QPrinterInfo_Location(const QPrinterInfo* self) {
	QString _ret = self->location();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QPrinterInfo_MakeAndModel(const QPrinterInfo* self) {
	QString _ret = self->makeAndModel();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

bool QPrinterInfo_IsNull(const QPrinterInfo* self) {
	return self->isNull();
}

bool QPrinterInfo_IsDefault(const QPrinterInfo* self) {
	return self->isDefault();
}

bool QPrinterInfo_IsRemote(const QPrinterInfo* self) {
	return self->isRemote();
}

int QPrinterInfo_State(const QPrinterInfo* self) {
	return self->state();
}

libqt_list QPrinterInfo_SupportedPageSizes(const QPrinterInfo* self) {
	return self->supportedPageSizes();
}

QPageSize* QPrinterInfo_DefaultPageSize(const QPrinterInfo* self) {
	return new QPageSize(self->defaultPageSize());
}

bool QPrinterInfo_SupportsCustomPageSizes(const QPrinterInfo* self) {
	return self->supportsCustomPageSizes();
}

QPageSize* QPrinterInfo_MinimumPhysicalPageSize(const QPrinterInfo* self) {
	return new QPageSize(self->minimumPhysicalPageSize());
}

QPageSize* QPrinterInfo_MaximumPhysicalPageSize(const QPrinterInfo* self) {
	return new QPageSize(self->maximumPhysicalPageSize());
}

libqt_list QPrinterInfo_SupportedResolutions(const QPrinterInfo* self) {
	return self->supportedResolutions();
}

int QPrinterInfo_DefaultDuplexMode(const QPrinterInfo* self) {
	return self->defaultDuplexMode();
}

libqt_list QPrinterInfo_SupportedDuplexModes(const QPrinterInfo* self) {
	return self->supportedDuplexModes();
}

int QPrinterInfo_DefaultColorMode(const QPrinterInfo* self) {
	return self->defaultColorMode();
}

libqt_list QPrinterInfo_SupportedColorModes(const QPrinterInfo* self) {
	return self->supportedColorModes();
}

libqt_list QPrinterInfo_AvailablePrinterNames() {
	return QPrinterInfo::availablePrinterNames();
}

libqt_list QPrinterInfo_AvailablePrinters() {
	return QPrinterInfo::availablePrinters();
}

libqt_string QPrinterInfo_DefaultPrinterName() {
	QString _ret = QPrinterInfo::defaultPrinterName();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QPrinterInfo* QPrinterInfo_DefaultPrinter() {
	return new QPrinterInfo(QPrinterInfo::defaultPrinter());
}

QPrinterInfo* QPrinterInfo_PrinterInfo(const libqt_string printerName) {
	return new QPrinterInfo(QPrinterInfo::printerInfo(QString::fromUtf8(printerName.data, printerName.len)));
}

void QPrinterInfo_Delete(QPrinterInfo* self) {
    delete self;
}

